pkg/controllers/mcs/multiclusterservice_controller.go

```diff
@@ -0,0 +1,620 @@
+				Namespace: mcs.Namespace,
+				Name:      mcs.Name,
+			}})
+		case *clusterv1alpha1.Cluster:
+			mcsLists := &networkingv1alpha1.MultiClusterServiceList{}
+			err := c.Client.List(context.TODO(), mcsLists)
+			if err != nil {
+				return nil
+			}
+			for _, item := range mcsLists.Items {
+				if len(item.Spec.Range.ClusterNames) == 0 {
+					requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{
+						Namespace: item.Namespace,
+						Name:      item.Name,
+					}})
+					continue
+				}
+				for _, clusterName := range item.Spec.Range.ClusterNames {
+					if clusterName != t.Name {
+						continue
+					}
+					requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{
+						Namespace: item.Namespace,
+						Name:      item.Name,
+					}})
+				}
+			}
+		default:
+			err := c.Client.Get(context.TODO(), types.NamespacedName{Namespace: t.GetNamespace(), Name: t.GetName()}, mcs)
+			if err != nil {
+				return nil
+			}
+			requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{
+				Namespace: mcs.Namespace,
+				Name:      mcs.Name,
+			}})
+		}
+		return
+	}
+}
+
+// SetupWithManager creates a controller and register to controller manager.
+func (c *MultiClusterServiceController) SetupWithManager(mgr controllerruntime.Manager) error {
+	return utilerrors.NewAggregate([]error{
+		mgr.Add(c),
+		controllerruntime.NewControllerManagedBy(mgr).
+			For(&networkingv1alpha1.MultiClusterService{}).
+			Watches(&source.Kind{Type: &corev1.Service{}}, handler.EnqueueRequestsFromMapFunc(c.newMultiClusterServiceFunc())).
+			Watches(&source.Kind{Type: &workv1alpha2.ResourceBinding{}}, handler.EnqueueRequestsFromMapFunc(c.newMultiClusterServiceFunc()),
+				// Only observe the rb of the svc.
+				builder.WithPredicates(predicate.Funcs{
+					CreateFunc: func(createEvent event.CreateEvent) bool {
+						newRB := createEvent.Object.(*workv1alpha2.ResourceBinding)
+						if newRB.Spec.Resource.APIVersion != corev1.SchemeGroupVersion.String() || newRB.Spec.Resource.Kind != util.ServiceKind {
+							return false
+						}
+						return true
+					},
+					UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+						oldRB := updateEvent.ObjectOld.(*workv1alpha2.ResourceBinding)
+						newRB := updateEvent.ObjectNew.(*workv1alpha2.ResourceBinding)
+						if newRB.Spec.Resource.APIVersion != corev1.SchemeGroupVersion.String() || newRB.Spec.Resource.Kind != util.ServiceKind ||
+							oldRB.Spec.Resource.APIVersion != corev1.SchemeGroupVersion.String() || oldRB.Spec.Resource.Kind != util.ServiceKind {
+							return false
+						}
+						return oldRB.Generation != newRB.Generation
+					},
+					DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+						return false
+					},
+				})).
+			Watches(&source.Kind{Type: &mcsv1alpha1.ServiceImport{}}, handler.EnqueueRequestsFromMapFunc(c.newMultiClusterServiceFunc()),
```

Comment:
`ServiceImport` and `ServiceExport` were created by us, do we still have to pay attention to their events?

Reply:
Yes, if `ServiceImport` or `ServiceExport` is accidentally deleted, they need to be recreated.

Reply:
Is it possible to ignore the creation and update events?

Reply:
Updated.